Drop moment.js for luxon.js

yubiuser opened the pull request:
**What does this PR aim to accomplish?:**

This PR initially only tried to replace https://github.com/dangrossman/daterangepicker which has been unmaintained for >5 years now. I found a active fork at https://github.com/Wernfried/daterangepicker which had the nice side effect that it replaced `moment.js` (also in [maintenance mode](https://momentjs.com/docs/)) mode with its successor `luxon` (https://moment.github.io/luxon). I swapped all the libraries and adjusted the code. Not all date/time format could be translated 1:1 but it's close. 

It was quite straight forward, but one `moment.js` function (`toHuman()`) did not work in `luxon.js` (with millisecond input it would not translate to human readable output but keep in miliseconds). There is a long-standing luxon bug (https://github.com/moment/luxon/issues/1134) which provided a nice workaround with a wrapper function. I added it, but it required to load `luxon.js` already in `header.lp` (opposed to `header_authenticated`).

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

yubiuser committed:
Swap moment.js for luxon.js and all js that depend on it

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Change js from moment to luxon style

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Replace toRelative with toHuman

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Fix tests

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Fix time/date style on settings page

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Fix formating in datetime() function

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Remove TTL remains from query log

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Fix querylog

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Fix tests

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Fix data format in daterange picker

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Fix daterange picker callback

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser commented:
Ready for review and testing

yubiuser commented:
I did not implement any of the new features https://github.com/Wernfried/daterangepicker would offer. This could be implemented in a follow-up PR.

PromoFaux commented:
While using this PR as a base for implementing the new constrained "All Time" range, I found a bug in the daterangepicker fork, and have submitted a PR upstream to get it fixed

https://github.com/Wernfried/daterangepicker/pull/1/files

yubiuser committed:
Update datarangepicker to 4.9.7

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Don't link calendars

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser commented:
Merge conflicts should resolve once we incorporate https://github.com/pi-hole/web/pull/3656

PromoFaux committed:
Refactor date range initialization to fetch earliest timestamp from API and update date picker accordingly.

We use the on-disk earliest timestamp to get the complete range.

Treat 0.0 response as NULL. Also get the in-memory timetamp, and then use whichever of the two timestamps is smallest (and non-zero)

Signed-off-by: Adam Warner <me@adamwarner.co.uk>
